packages/sundialsml/sundialsml.5.8.0p0/opam

```diff
@@ -0,0 +1,77 @@
+opam-version: "2.0"
+maintainer: "tim@tbrk.org"
+authors: [
+    "Timothy Bourke <tim@tbrk.org>"
+    "Jun Inoue <Jun.Lambda@gmail.com>"
+    "Marc Pouzet <Marc.Pouzet@ens.fr>"
+]
+homepage: "http://inria-parkas.github.io/sundialsml/"
+bug-reports: "https://github.com/inria-parkas/sundialsml/issues"
+dev-repo: "git+https://github.com/inria-parkas/sundialsml"
+doc: "http://inria-parkas.github.io/sundialsml/"
+tags: [
+    "numerical"
+    "simulation"
+    "mathematics"
+    "science"
+]
+license: "BSD-3-Clause"
+build: [
+    ["./configure" "--stubdir=%{stublibs}%/"
+		   "--libdir=%{lib}%/"
+		   "--docdir=%{doc}%/sundialsml/"
+		   "--os-distribution=%{os-distribution}%"]
+    [make "-j%{jobs}%"]
+    [make "doc"] {with-doc}
+]
+install: [
+  [make "install-findlib"]
+  [make "install-doc"] {with-doc}
+]
+remove: [
+    ["ocamlfind" "remove" "sundialsml"]
+    ["rm" "-rf" "%{doc}%/sundialsml"]
+]
+depends: [
+    "ocaml" {>= "4.03.0"}
+    "base-bigarray"
+    "ocamlfind" {build}
+    "conf-sundials" {>= "2" & build}
+]
+depopts: [
+    "mpi"
+]
+synopsis: "Interface to the Sundials suite of numerical solvers"
```

Comment:
```suggestion
synopsis: "Interface to the Sundials suite of numerical solvers"
conflicts: ["base-domains"]
```
It is not compatible with multicore:
```
# cc -I /home/opam/.opam/4.12/lib/ocaml  -O3 -Wall -Werror  -DNDEBUG=1 -fPIC -isystem /home/opam/.opam/4.12/lib/ocaml -isystem /usr/include/suitesparse/  -o nvectors/nvector_pthreads_ml.o -c nvectors/nvector_pthreads_ml.c
# In file included from sundials/sundials_ml.c:25:
# sundials/sundials_ml.c: In function 'sunml_sundials_malloc_value':
# sundials/sundials_ml.c:189:32: error: 'Caml_black' undeclared (first use in this function)
#   189 |     *block = Make_header(1, 0, Caml_black);
#       |                                ^~~~~~~~~~
# sundials/sundials_ml.c:189:32: note: each undeclared identifier is reported only once for each function it appears in
# sundials/sundials_ml.c: In function 'sunml_wrap_session_pointer':
# sundials/sundials_ml.c:225:35: error: assignment of member 'compare' in read-only object
#   225 |     Custom_ops_val(vmem)->compare = compare_session_pointers;
#       |                                   ^
# make[1]: *** [Makefile:118: sundials/sundials_ml.o] Error 1
# make[1]: *** Waiting for unfinished jobs....
# make[1]: Leaving directory '/home/opam/.opam/4.12/.opam-switch/build/sundialsml.5.8.0p0/src'
# make: *** [Makefile:8: all] Error 2
```

Reply:
Thanks for those suggestions @mseri. That's very helpful.
I'm not surprised that the library does not (yet) work against multicore OCaml. I'll have a look at this eventually.
I don't know what's going on with the problems on arm. That is surprising.